src/app/loadout/fashion/FashionDrawer.tsx

```diff
@@ -0,0 +1,433 @@
+          <FashionItem
+            key={bucketHash}
+            classType={classType}
+            bucketHash={bucketHash}
+            item={armorItemsByBucketHash[bucketHash]}
+            mods={modsByBucket[bucketHash]}
+            onPickPlug={setPickPlug}
+            onRemovePlug={handleRemovePlug}
+          />
+        ))}
+        {!isPhonePortrait && (
+          <div className={clsx(styles.verticalButtons, styles.right)}>{rightButtons}</div>
+        )}
+      </div>
+      {isPhonePortrait && (
+        <div className={styles.buttons}>
+          {leftButtons}
+          {rightButtons}
+        </div>
+      )}
+      {pickPlug &&
+        ReactDOM.createPortal(
+          <SocketDetails
+            item={pickPlug.item}
+            allowInsertPlug={false}
+            socket={pickPlug.socket}
+            onClose={() => setPickPlug(undefined)}
+            onPlugSelected={handlePlugSelected}
+          />,
+          document.body
+        )}
+    </Sheet>
+  );
+}
+
+function FashionItem({
+  item,
+  classType,
+  bucketHash,
+  mods = [],
+  onPickPlug,
+  onRemovePlug,
+}: {
+  item?: DimLoadoutItem;
+  classType: DestinyClass;
+  bucketHash: number;
+  mods?: number[];
+  onPickPlug(params: PickPlugState): void;
+  onRemovePlug(bucketHash: number, modHash: number): void;
+}) {
+  const defs = useD2Definitions()!;
+  const isShader = (m: number) =>
+    defs.InventoryItem.get(m)?.plug?.plugCategoryHash === PlugCategoryHashes.Shader;
+  const shader = mods.find(isShader);
+  const ornament = mods.find((m) => !isShader(m));
+
+  const shaderItem = shader ? defs.InventoryItem.get(shader) : undefined;
+  const ornamentItem = ornament ? defs.InventoryItem.get(ornament) : undefined;
+
+  const allItems = useSelector(allItemsSelector);
+  const unlockedPlugSetItems = useSelector(unlockedPlugSetItemsSelector);
+
+  // TODO: is this really the best way to do this? we just default to the equipped item, but that may be an exotic
+  const exampleItem =
+    item ??
+    allItems.find(
+      (i) =>
+        i.bucket.hash === bucketHash &&
+        i.power &&
+        i.tier === 'Legendary' &&
+        i.classType === classType
+    ) ??
+    allItems.find(
+      (i) =>
+        i.bucket.hash === bucketHash && i.power && i.tier !== 'Exotic' && i.classType === classType
+    );
+
+  if (!exampleItem) {
+    return null;
+  }
+
+  const isShaderSocket = (s: DimSocket) =>
+    defs.SocketType.get(s.socketDefinition.socketTypeHash)?.plugWhitelist.some(
+      (pw) => pw.categoryHash === PlugCategoryHashes.Shader
+    );
+  const cosmeticSockets = getSocketsByCategoryHash(
+    exampleItem.sockets,
+    SocketCategoryHashes.ArmorCosmetics
+  );
+  const shaderSocket = cosmeticSockets.find(isShaderSocket);
+  const ornamentSocket = cosmeticSockets.find((s) => !isShaderSocket(s));
+
+  const defaultShader = defs.InventoryItem.get(DEFAULT_SHADER);
+  const defaultOrnament = defs.InventoryItem.get(DEFAULT_ORNAMENTS[0]);
```

Comment:
This (and the other usage) isn't actually the correct default ornament, this one says "Restores your weapon to its default appearance" (the correct one is `702981643` / `DEFAULT_ORNAMENTS[2]`). Purely a visual bug though.